activation/handler.go

```diff
@@ -513,24 +518,26 @@ func (h *Handler) handleGossipAtx(ctx context.Context, peer p2p.Peer, msg []byte
 		return fmt.Errorf("nil nipst in gossip for atx %s", atx.ShortString())
 	}
 
-	h.registerHashes(&atx, peer)
+	h.registerHashes(&atx, peer) // TODO(mafa): should this be done after the ATX was validated?
```

Comment:
Clarify.

Reply:
it needs to be registered before handler calls fetcher. this is the way syncer keeps track of which peers has missing objects